enzyme/test/TypeAnalysis/smax.ll

```diff
@@ -0,0 +1,35 @@
+; RUN: if [ %llvmver -lt 16 ]; then %opt < %s %loadEnzyme -opaque-pointers -print-type-analysis -type-analysis-func=smax -o /dev/null | FileCheck %s; fi
```

Comment:
@wsmoses How do you usually not run test for older versions?

Reply:
So this test shouldn't need a ptr regardles.

E.g. a minimal test would be:


```
define i64 f(i64 a0, i64 b0) {
   c0 = call i53 llvm.smax(a, b)
   a = call i64 @run()
   b = call i64 @run()
   c = call i53 llvm.smax(a, b)
   ret c;
}
```

Arguments and returns are given reasonable types (aka initialied to int here). This the c0 call will test that we forward propagate from args to return, and the a/b/c test will test that we backwards propagate from return to args.

In any case the way we version gate is the `if [ %llvmver -lt 16 ]; then` you can see in the run file. That is a if version < 16 use old pass manager tests, and the line below (presently unconditionally) runs new pass manager versions